USM.Test/AlgorithmTest.cs

```diff
@@ -0,0 +1,509 @@
+            Course[] AllCourses = Classes();
+
+            // choose the courses to be tested
+            Course CSE20 = AllCourses[0];
+            Course CSE23 = AllCourses[3];
+            Course[] courses = { CSE20, CSE23 };
+
+            //call the algorithm
+            List<List<Section>> returned = sch.FindScheduleForClasses(courses);
+            int expectedNum = 1;
+            int actualNum = returned.Count;
+
+            //two schedules
+            Assert.Equal(expectedNum, actualNum);
+        }
+
+        //no conflict on dif day different time
+        [Fact]
+        public void TestNoConflictDifDaySameTime()
+        {
+            //use the course above
+            ScheduleRepo sch = new ScheduleRepo();
+            Course[] AllCourses = Classes();
+
+            // choose the courses to be tested
+            Course CSE20 = AllCourses[0];
+            Course CSE24 = AllCourses[4];
+            Course[] courses = { CSE20, CSE24 };
+
+            //call the algorithm
+            List<List<Section>> returned = sch.FindScheduleForClasses(courses);
+            int expectedNum = 1;
+            int actualNum = returned.Count;
+
+            //two schedules
+            Assert.Equal(expectedNum, actualNum);
+        }
+
+        //no conflict on dif day same time
+        [Fact]
+        public void TestNoConflictDifDayDifTime()
+        {
+            //use the course above
+            ScheduleRepo sch = new ScheduleRepo();
+            Course[] AllCourses = Classes();
+
+            // choose the courses to be tested
+            Course CSE20 = AllCourses[0];
+            Course CSE25 = AllCourses[5];
+            Course[] courses = { CSE20, CSE25 };
+
+            //call the algorithm
+            List<List<Section>> returned = sch.FindScheduleForClasses(courses);
+            int expectedNum = 1;
+            int actualNum = returned.Count;
+
+            //two schedules
+            Assert.Equal(expectedNum, actualNum);
+        }
+
+        // no conflict with two discussions in one class
+        [Fact]
+        public void TestNoConflictWithDis1()
+        {
+            //use the course above
+            ScheduleRepo sch = new ScheduleRepo();
+            Course[] AllCourses = Classes();
+
+            // choose the courses to be tested
+            Course CSE20 = AllCourses.First(c => c.CourseAbbreviation == "CSE20");
+            Course CSE26 = AllCourses.First(c => c.CourseAbbreviation == "CSE26");
+            Course[] courses = { CSE20, CSE26 };
+
+            //call the algorithm
+            List<List<Section>> returned = sch.FindScheduleForClasses(courses);
+            int expectedNum = 2;
+            int actualNum = returned.Count;
+
+            //two schedules
+            List<Section> ls0 = returned[0];
+            Section dis1 = ls0[1];
+            List<Section> ls1 = returned[1];
+            Section dis2 = ls1[1];
+            Debug.WriteLine("first dis is "+dis1.Course.CourseAbbreviation);
```

Comment:
C# has string interpolation. Use: $"First discussion is: {dis1.course...}" instead.